### PR TITLE
Feature/1 basic movement controller

### PR DESCRIPTION
**Pull request by RyanJST**
The player controller should be separated across different scripts to ensure that the code for movement/shooting is kept independent and clean. This development task is related to the player movement controller. Develop basic first person keyboard and mouse controls that support the following actions:

- Horizontal character movement using the keyboard (default: WASD). Some level of movement acceleration/deceleration should be incorporated.

- Free look using mouse input. Vertical look angle constraints should be used to prevent the camera from breaking (unless another solution can prevent that as well)

- Jumping using the keyboard (default: spacebar)

- Crouching using the keyboard (default: left control)

- "Walk" mode using the keyboard (default: left shift)

- Basic collision resolution for levels and obstacles (player stops when making contact)

- Variables such as acceleration of movement, jump height, crouch height, walk speed, run speed, vertical look angle constraints, and mouse look sensitivity should all be easy to manipulate so that play-testing different settings is simple to do.

### CODE REVIEW

File: Assets/Scripts/Player/MouseLookController.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MouseLookController : MonoBehaviour
```
Comment (overseer25) [inline Assets/Scripts/Player/MouseLookController.cs]: Why does this file exist?

File: Assets/Scripts/MovementController.cs.meta [hunk 2]
```diff
@@ -0,0 +1,11 @@
+fileFormatVersion: 2
```
Comment (overseer25) [inline Assets/Scripts/MovementController.cs.meta]: What is this MovementController script? It is showing no changes for the cs code for it in this PR? Is it different from the BaseMovementController?

File: Assets/Scripts/Player/PlayerMovementController.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            //velocity.y = JumpHeight;
+            jumping = true;
+            startJumpHeight = transform.position.y;
+            grounded = false;
+        }
+    }
+
+    protected override void Move()
+    {
+        acceleration = Vector3.zero;
```
Comment (overseer25) [inline Assets/Scripts/Player/PlayerMovementController.cs]: Acceleration will need to be revisited, as it does not seem to work in this current iteration.

Comment (packerbacker8) [reply Assets/Scripts/Player/PlayerMovementController.cs]: acceleration aint a thing in this current version. intentionally

**Pull request merged**